### PR TITLE
New major version with several enhancements

### PR DESCRIPTION
**Pull request by mlopezFC**
New major version with the following features:
* Use latest wc-chatbot web component version
* Add support for customizing header and footer
* Complete rewrite of the internal chat history, based on vaadin components instead of wc-bubble
* Support for virtual list to avoid performance issues with large chats
* Fixes visual issues when reattaching, maintaining the state
* Support for displaying who is typing and also if a message is being generated (dancing dots)

## Release Notes

- **New Features**
  - Major version update to 2.0.0, enhancing overall functionality.
  - Improved chat functionality with real-time message updates and typing indicators.
  - New chat message structure for better user experience.
  - Added loading animations and typing indicators for enhanced visual feedback.
  - Introduced lazy loading capabilities for chat messages in the demo.
  - Enhanced demo with a "Chat With Thinking" button for better user interaction.
  - New styles for chat messages and loading indicators to improve visual presentation.

- **Bug Fixes**
  - Upgraded dependencies for improved performance and stability.

- **Documentation**
  - Enhanced user interaction in the demo with new features and updated greetings.

### CODE REVIEW

File: src/main/java/com/flowingcode/vaadin/addons/chatassistant/ChatMessage.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      this.setUserName(message.getSender().getName());
+      if (message.getSender().getAvatar()!=null) {
+        this.setUserImg(message.getSender().getAvatar());
+      }
+    }
+    if (message.getMessageTime()!=null) {
+      String formattedTime = message.getMessageTime().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
+      this.setTime(formattedTime);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (mlopezFC) [reply src/main/java/com/flowingcode/vaadin/addons/chatassistant/ChatMessage.java]: Done ✔️

File: src/main/java/com/flowingcode/vaadin/addons/chatassistant/ChatAssistant.java [hunk 2]
```diff
@@ -38,75 +52,259 @@
...
+  /**
+   * Clears the text shown over the message input to indicate that someone is typing
+   */
+  public void clearWhoIsTyping() {
+    this.whoIsTyping.setText(null);
+    this.whoIsTyping.setVisible(false);
+  }
+  
+  /**
+   * Sets the SubmitListener that will be notified when the user submits a message on the underlying messageInput
```
Comment (paodb) [inline src/main/java/com/flowingcode/vaadin/addons/chatassistant/ChatAssistant.java:151]: Description should end with a dot. There should be a empty line between the description and the tags. Param and return tags should have a brief description also, and shouldn't end with a dot. See documentation https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#format

File: src/main/java/com/flowingcode/vaadin/addons/chatassistant/ChatAssistant.java [hunk 3]
```diff
@@ -38,75 +52,259 @@
...
+  private MessageInput messageInput;
+  private Span whoIsTyping;
+  private boolean minimized = false;
+  private Registration defaultSubmitListenerRegistration;
+
+  public ChatAssistant() {
+    this(new ArrayList<>());
+  }
+
+  public ChatAssistant(List<Message> messages) {
```
Comment (paodb) [inline src/main/java/com/flowingcode/vaadin/addons/chatassistant/ChatAssistant.java:75]: Add Javadoc

File: src/main/java/com/flowingcode/vaadin/addons/chatassistant/ChatAssistant.java [hunk 4]
```diff
@@ -38,75 +52,259 @@
...
+  private Component headerComponent;
+  private Component footerComponent;
+  private VirtualList<Message> content = new VirtualList<>();
+  private List<Message> messages;
+  private MessageInput messageInput;
+  private Span whoIsTyping;
+  private boolean minimized = false;
+  private Registration defaultSubmitListenerRegistration;
+
+  public ChatAssistant() {
```
Comment (paodb) [inline src/main/java/com/flowingcode/vaadin/addons/chatassistant/ChatAssistant.java:71]: Add Javadoc

**Review by paodb — CHANGES_REQUESTED** [review]
Left some comments regarding Javadocs. Please check documentation is complete. Follow shared example.

**Comment by paodb** [comment]
Besides the requested changes, I tested the branch and the upgrade seems okay to me.

File: src/main/java/com/flowingcode/vaadin/addons/chatassistant/ChatAssistant.java (hunk 4 above)
Comment (mlopezFC) [reply src/main/java/com/flowingcode/vaadin/addons/chatassistant/ChatAssistant.java:74]: Done ✔️

**Review by paodb — APPROVED** [review]
LGTM

**Pull request merged**